File: plugins/geolocation/android/src/main/java/io/letary/lecodes/plugins/geolocation/GeolocationPlugin.kt
//
//  GeolocationPlugin.kt — lecodes-plugins/geolocation
//
//  The "geolocation" registerService plugin — backs the SDK `Geolocation` global.
//  Extracted from the SDK for the same reason as on iOS (there: linking CoreLocation
//  flags every consumer; here: the location permissions in the merged manifest do).
//

package io.letary.lecodes.plugins.geolocation

import android.content.Context
import io.letary.lecodes.LecodesEngine

object GeolocationPlugin {

    const val PLUGIN_ID = "geolocation"

    fun register(engine: LecodesEngine, context: Context) {
        val appContext = context.applicationContext
        engine.registerService("geolocation") { _, channel ->
            GeolocationService(engine, appContext, channel).instance()
        }
    }
}

File: plugins/geolocation/ios/GeolocationPlugin.swift
//
//  GeolocationPlugin.swift — lecodes-plugins/geolocation
//
//  The "geolocation" registerService plugin: the SDK's Geolocation wrapper rides the
//  generic service channel, no per-plugin bridge methods. Extracted from the SDK so that
//  only apps that install it link CoreLocation (Apple flags location usage from the mere
//  presence of CLLocationManager symbols); the manifest merges the Info.plist usage string.
//

import Foundation
import LeCodesSDK

public enum LeCodesGeolocationPlugin {

    public static let pluginId = "geolocation"

    public static func register(in engine: LeCodesEngine) {
        engine.registerService("geolocation") { _, channel in
            GeolocationService(channel)
        }
    }
}
